libs/components/src/lib/toggletip/toggletip.spec.ts

```diff
@@ -0,0 +1,162 @@
+		}));
+
+	beforeEach(async () => {
+		element = fixture(
+			`<${COMPONENT_TAG}></${COMPONENT_TAG}>`
+		) as Toggletip;
+
+		anchor = fixture(
+			'<vwc-button id="anchorButton"></vwc-button>', ADD_TEMPLATE_TO_FIXTURE
+		) as Button;
+	});
+
+	describe('basic', () => {
+		it('should be initialized as a vwc-toggletip', async () => {
+			expect(toggletipDefinition()).toBeInstanceOf(FoundationElementRegistry);
+			expect(element).toBeInstanceOf(Toggletip);
+
+			expect(element.open).toBe(false);
+			expect(element.alternate).toBe(false);
+			expect(element.headline).toBeUndefined();
+			expect(element.placement).toBe('right');
+		});
+	});
+
+	describe('open/close', () => {
+		it('should open when its anchor is clicked', async () => {
+			element.anchor = 'anchorButton';
+			await elementUpdated(element);
+
+			expect(element.open).toBe(false);
+
+			anchor.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(true);
+		});
+
+		it('should remain open when clicked inside', async () => {
+			element.anchor = 'anchorButton';
+			element.open = true;
+			await elementUpdated(element);
+
+			element.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(true);
+		});
+
+		it('should close when clicked outside', async () => {
+			element.anchor = 'anchorButton';
+			element.open = true;
+			await elementUpdated(element);
+
+			document.body.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(false);
+		});
+
+		it('should close when Escape is pressed', async () => {
+			element.anchor = 'anchorButton';
+			element.open = true;
+			await elementUpdated(element);
+
+			document.body.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape', bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(false);
+		});
+	
+	});
+
+	describe('anchor', () => {
+		it('should accept an HTMLElement as anchor', async () => {
+			element.anchor = anchor;
+			await elementUpdated(element);
+
+			anchor.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+
+			expect(element.open).toEqual(true);
+		});
+
+		it('should open correctly when its anchor is changed', async () => {
+			const anchor2 = fixture(
+				'<vwc-button id="anchorButton2"></vwc-button>', ADD_TEMPLATE_TO_FIXTURE
+			) as Button;
+	
+			element.anchor = 'anchorButton';
+			await elementUpdated(element);
+
+			element.anchor = 'anchorButton2';
+			await elementUpdated(element);
+			expect(element.open).toEqual(false);
+
+			anchor.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+			expect(element.open).toEqual(false);
+
+			anchor2.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+			await elementUpdated(element);
+			expect(element.open).toEqual(true);
+		});
```

Comment:
There are two functionalities tested. Splitting is advised.

```suggestion
		it('should set listener to new anchor when anchor is changed', async () => {
			const anchor2 = fixture(
				'<vwc-button id="anchorButton2"></vwc-button>', ADD_TEMPLATE_TO_FIXTURE
			) as Button;
	
			element.anchor = 'anchorButton';
			await elementUpdated(element);

			element.anchor = 'anchorButton2';
			await elementUpdated(element);
			element.open = false;

			anchor2.dispatchEvent(new MouseEvent('click', {bubbles: true}));
			await elementUpdated(element);
			expect(element.open).toEqual(true);
		});
		
		it('should remove listener on old anchor when anchor is changed', async () => {
			const anchor2 = fixture(
				'<vwc-button id="anchorButton2"></vwc-button>', ADD_TEMPLATE_TO_FIXTURE
			) as Button;
	
			element.anchor = 'anchorButton';
			await elementUpdated(element);

			element.anchor = 'anchorButton2';
			await elementUpdated(element);
			element.open = false;

			anchor.dispatchEvent(new MouseEvent('click', {bubbles: true}));
			await elementUpdated(element);
			expect(element.open).toEqual(false);
		});
```